### ISSUE TITLE
encryption: false option?

### ISSUE THREAD

**Issue by gmaclennan** [holepunchto/hyperswarm-secret-stream#21]
Any interest in adding a `{ encrypted: false }` option? I could work on a PR.

**Use cases:**

- Inspecting network traffic to debug code
- Using secret-stream over an existing encrypted connection

My specific use-case is hypercore replication over a secure websocket (e.g. TLS). The secret-stream handshake to confirm public keys is still useful, but the encryption/decryption adds CPU overhead and is not needed in this use-case.

I understand that this is an edge-case, so I can implement as a project-specific patch if you don't want to add to the API here.

**Comment by mafintosh** [comment]
if trivial, sure, go for it. Note that hypercore relies on the noise-handshake setup so the only thing you can safely disable is the transport encryption. Also the option should be something like `unsafeDisableEncryption: true`

**Comment by mafintosh** [comment]
You would need to impl noops of Push and Pull here so it doesnt impact any of the other code at runtime

**Comment by mafintosh** [comment]
https://github.com/holepunchto/hyperswarm-secret-stream/blob/master/index.js#L1

**Comment by av8ta** [comment]
I have the second use case mentioned above:

> Using secret-stream over an existing encrypted connection

It's in a browser environment trying to replicate hypercores and is erroring on the noise handshake. 

@gmaclennan did you work on this / advice to give?

@mafintosh is there another solution for replicating hypercores without noise?

**Comment by gmaclennan** [comment]
> @gmaclennan did you work on this / advice to give?

No, I haven't yet. So far I've just been piping the whole encrypted noise stream over websocket, but it's still at the prototype stage so I haven't looked at the performance gains from removing the encryption (because the websocket is over TLS).

**Comment by gmaclennan** [comment]
Created a PR to implement this: https://github.com/holepunchto/hyperswarm-secret-stream/pull/25

**Comment by av8ta** [comment]
Thanks @gmaclennan that was quick! I've tried it out your PR in node and it works great. Still not working for me in webview though - it's the noise handshake that's failing as noted by others in this issue https://github.com/holepunchto/hyperswarm-secret-stream/issues/13

**Comment by av8ta** [comment]
Found an issue tracking browser compatibility. There's just one task remaining of 22 tasks!

https://github.com/hypercore-protocol/hypercore-next/issues/53

https://github.com/sodium-friends/sodium-javascript/pull/65

@chm-diederichs is there much involved to get this across the line? :pray:

**Comment by chm-diederichs** [comment]
Hey, possibly not too much work. I will take a look when I can, but I'm a little swamped at the moment

I have the methods implemented on [this branch](https://github.com/chm-diederichs/sodium-javascript/tree/curve) here though, if you want to test in the meantime. Just be aware it's an experimental branch.

**Comment by gmaclennan** [comment]
Sorry for the noise, just checking the view of the maintainers on this issue and the implementation in https://github.com/holepunchto/hyperswarm-secret-stream/pull/25. I understand there is the separate question of browser compatibility, but this feature and solution does not need to depend on that.

We can always just patch / fork, but it's easier to maintain into the future for us if this is integrated here. See the PR, there is significant overhead saved by turning off encryption.

**Comment by mafintosh** [comment]
Hi @gmaclennan . For now rely on the branch. We are pushing a lot of changes through the stack atm with rocks, so dont wanna rock the boat too much and land this while that happens. Feature wise, ok to have the flag and it will land eventually
